Author: Jamie Strandboge <jamie@ubuntu.com>
Description: suppress iptables warnings about legacy rules being present
 in root functional tests
Index: ufw-0.36/tests/root/bugs/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/bugs/runtest.sh
+++ ufw-0.36/tests/root/bugs/runtest.sh
@@ -24,7 +24,7 @@ echo "Bug #247352" >> $TESTTMP/result
 do_cmd "0" --dry-run allow http/tcp
 grep -A2 "tuple" $TESTCONFIG/user.rules >> $TESTTMP/result
 echo "iptables -L -n:" >> $TESTTMP/result
-iptables -L -n | grep -A1 "80" >> $TESTTMP/result 2>&1
+iptables -L -n 2>/dev/null | grep -A1 "80" >> $TESTTMP/result
 do_cmd "0" delete allow http/tcp
 grep -A2 "tuple" $TESTCONFIG/user.rules >> $TESTTMP/result
 
@@ -33,12 +33,12 @@ echo "Setting IPV6 to no" >> $TESTTMP/re
 sed -i "s/IPV6=.*/IPV6=no/" $TESTPATH/etc/default/ufw
 do_cmd "0" nostats disable
 echo "/lib/ufw/ufw-init flush-all:" >> $TESTTMP/result
-$TESTSTATE/ufw-init flush-all >> $TESTTMP/result 2>&1
+$TESTSTATE/ufw-init flush-all 2>/dev/null >> $TESTTMP/result
 do_cmd "0" nostats enable
 echo "/lib/ufw/ufw-init force-reload:" >> $TESTTMP/result
-$TESTSTATE/ufw-init force-reload >> $TESTTMP/result 2>&1
+$TESTSTATE/ufw-init force-reload 2>/dev/null >> $TESTTMP/result
 echo "ip6tables -L -n:" >> $TESTTMP/result
-ip6tables -L -n >> $TESTTMP/result 2>&1
+ip6tables -L -n 2>/dev/null >> $TESTTMP/result
 
 echo "Bug #260881" >> $TESTTMP/result
 echo "Setting IPV6 to no" >> $TESTTMP/result
@@ -48,10 +48,10 @@ do_cmd "0" nostats enable
 do_cmd "0"  allow Apache
 do_cmd "0"  delete deny Apache
 echo "iptables -L -n:" >> $TESTTMP/result
-iptables -L -n | grep -A1 "80" >> $TESTTMP/result 2>&1
+iptables -L -n 2>/dev/null | grep -A1 "80" >> $TESTTMP/result
 do_cmd "0"  delete allow Apache
 echo "iptables -L -n:" >> $TESTTMP/result
-iptables -L -n | grep -A1 "80" >> $TESTTMP/result 2>&1
+iptables -L -n 2>/dev/null | grep -A1 "80" >> $TESTTMP/result
 
 echo "Bug #263308" >> $TESTTMP/result
 echo "Setting IPV6 to yes" >> $TESTTMP/result
@@ -138,13 +138,13 @@ for ipv6 in yes no ; do
             do_cmd "0" null logging $i
         fi
         do_cmd "0" null enable
-        iptables-save | grep '^-' > $TESTTMP/ipt.enable
-        ip6tables-save | grep '^-' > $TESTTMP/ip6t.enable
+        iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.enable
+        ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.enable
 
         do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init stop
         do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init start
-        iptables-save | grep '^-' > $TESTTMP/ipt.start
-        ip6tables-save | grep '^-' > $TESTTMP/ip6t.start
+        iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.start
+        ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.start
 
         diff $TESTTMP/ipt.enable $TESTTMP/ipt.start || {
             echo "'ufw enable' and 'ufw-init start' are different for loglevel '$i'"
@@ -166,7 +166,7 @@ for i in low on medium high full off off
     if [ "$i" = "off" ]; then
         e="1"
     fi
-    iptables-save | grep -q 'UFW LIMIT BLOCK' $TESTCONFIG/user.rules
+    iptables-save 2>/dev/null | grep -q 'UFW LIMIT BLOCK' $TESTCONFIG/user.rules
     rc="$?"
     if [ "$rc" != "$e" ]; then
         echo "$i: got '$rc', expected '$e'"
@@ -181,7 +181,7 @@ do_cmd "0" nostats enable
 for b in INPUT OUTPUT FORWARD; do
     suffix=`echo $b | tr [A-Z] [a-z]`
     echo "$count: iptables -L $b -n | egrep -q 'ufw-after-logging-$suffix'" >> $TESTTMP/result
-    iptables -L "$b" -n | egrep -q "ufw-after-logging-$suffix" || {
+    iptables -L "$b" -n 2>/dev/null | egrep -q "ufw-after-logging-$suffix" || {
         echo "'iptables -L $b -n' does not contain 'ufw-after-logging-$suffix'"
         exit 1
     }
Index: ufw-0.36/tests/root/live/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/live/runtest.sh
+++ ufw-0.36/tests/root/live/runtest.sh
@@ -110,7 +110,7 @@ do
         echo iptables -I INPUT -j ACCEPT -m comment --comment $str >> $TESTTMP/result
         iptables -I INPUT -j ACCEPT -m comment --comment $str >> $TESTTMP/result
         do_cmd "0" nostats enable
-        iptables -n -L INPUT | grep "$str" >> $TESTTMP/result
+        iptables -n -L INPUT 2>/dev/null | grep "$str" >> $TESTTMP/result
         iptables -D INPUT -j ACCEPT -m comment --comment $str 2>/dev/null
 done
 
@@ -213,21 +213,21 @@ do_cmd "0" nostats disable
 do_cmd "0" nostats allow 23/tcp
 do_cmd "0" nostats logging medium
 do_cmd "0" null enable
-iptables-save | grep '^-' > $TESTTMP/ipt.enable
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.enable
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.enable
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.enable
 
 do_cmd "0" null disable
-iptables-save | grep '^-' > $TESTTMP/ipt.disable
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.disable
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.disable
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.disable
 
 sed -i 's/^ENABLED=no/ENABLED=yes/' $TESTPATH/etc/ufw/ufw.conf
 do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init start
-iptables-save | grep '^-' > $TESTTMP/ipt.start
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.start
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.start
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.start
 
 do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init stop
-iptables-save | grep '^-' > $TESTTMP/ipt.stop
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.stop
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.stop
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.stop
 
 diff $TESTTMP/ipt.enable $TESTTMP/ipt.start || {
 	echo "'ufw enable' and 'ufw-init start' are different"
@@ -264,7 +264,7 @@ for l in off on low medium high full; do
         for c in before-logging before after after-logging reject track ; do
             suffix=`echo $b | tr [A-Z] [a-z]`
             echo "$count: iptables -L $b -n | egrep -q 'ufw-$c-$suffix'" >> $TESTTMP/result
-            iptables -L $b -n | egrep -q "ufw-$c-$suffix" || {
+            iptables -L $b -n 2>/dev/null | egrep -q "ufw-$c-$suffix" || {
                 echo "'iptables -L $b -n' does not contain 'ufw-$c-$suffix'"
                 exit 1
             }
@@ -283,7 +283,7 @@ for l in off on low medium high full; do
     do_cmd "0" nostats enable
     for c in logging-deny not-local user-forward user-input user-output skip-to-policy-input ; do
         echo "$count: ! iptables -L ufw-$c -n | egrep -q '0 references'" >> $TESTTMP/result
-        iptables -L ufw-$c -n | egrep -q '0 references' && {
+        iptables -L ufw-$c -n 2>/dev/null | egrep -q '0 references' && {
             echo "'iptables -L ufw-user-input -n' had 0 references"
             exit 1
         }
@@ -293,7 +293,7 @@ for l in off on low medium high full; do
     done
     for c in logging-allow user-limit user-limit-accept user-logging-forward user-logging-input user-logging-output skip-to-policy-output skip-to-policy-forward ; do
         echo "$count: iptables -L ufw-$c -n | egrep -q '0 references'" >> $TESTTMP/result
-        iptables -L ufw-$c -n | egrep -q '0 references' || {
+        iptables -L ufw-$c -n 2>/dev/null | egrep -q '0 references' || {
             echo "'iptables -L ufw-user-input -n' had more than 0 references"
             exit 1
         }
Index: ufw-0.36/tests/root/live_route/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/live_route/runtest.sh
+++ ufw-0.36/tests/root/live_route/runtest.sh
@@ -99,7 +99,7 @@ do
         echo iptables -I FORWARD -j ACCEPT -m comment --comment $str >> $TESTTMP/result
         iptables -I FORWARD -j ACCEPT -m comment --comment $str >> $TESTTMP/result
         do_cmd "0" nostats enable
-        iptables -n -L FORWARD | grep "$str" >> $TESTTMP/result
+        iptables -n -L FORWARD 2>/dev/null | grep "$str" >> $TESTTMP/result
         iptables -D FORWARD -j ACCEPT -m comment --comment $str 2>/dev/null
 done
 
@@ -212,21 +212,21 @@ do_cmd "0" nostats disable
 do_cmd "0" nostats route allow 23/tcp
 do_cmd "0" nostats logging medium
 do_cmd "0" null enable
-iptables-save | grep '^-' > $TESTTMP/ipt.enable
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.enable
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.enable
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.enable
 
 do_cmd "0" null disable
-iptables-save | grep '^-' > $TESTTMP/ipt.disable
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.disable
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.disable
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.disable
 
 sed -i 's/^ENABLED=no/ENABLED=yes/' $TESTPATH/etc/ufw/ufw.conf
 do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init start
-iptables-save | grep '^-' > $TESTTMP/ipt.start
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.start
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.start
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.start
 
 do_extcmd "0" null $TESTPATH/lib/ufw/ufw-init stop
-iptables-save | grep '^-' > $TESTTMP/ipt.stop
-ip6tables-save | grep '^-' > $TESTTMP/ip6t.stop
+iptables-save 2>/dev/null | grep '^-' > $TESTTMP/ipt.stop
+ip6tables-save 2>/dev/null | grep '^-' > $TESTTMP/ip6t.stop
 
 diff $TESTTMP/ipt.enable $TESTTMP/ipt.start || {
 	echo "'ufw enable' and 'ufw-init start' are different"
Index: ufw-0.36/tests/root/live_apps/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/live_apps/runtest.sh
+++ ufw-0.36/tests/root/live_apps/runtest.sh
@@ -169,12 +169,12 @@ for ipv6 in no yes ; do
     cat $TESTCONFIG/user.rules >> $TESTTMP/result
     cat $TESTCONFIG/user6.rules >> $TESTTMP/result
 
-    iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.1
-    ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.1
+    iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.1
+    ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.1
     do_cmd "0" nostats disable
     do_cmd "0" nostats enable
-    iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.2
-    ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.2
+    iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.2
+    ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.2
     diff $TESTTMP/save.1 $TESTTMP/save.2 || {
         echo "ip(6)tables-restore different for '$i'"
         exit 1
@@ -197,12 +197,12 @@ for ipv6 in no yes ; do
     cat $TESTCONFIG/user.rules >> $TESTTMP/result
     cat $TESTCONFIG/user6.rules >> $TESTTMP/result
 
-    iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.1
-    ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.1
+    iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.1
+    ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.1
     do_cmd "0" nostats disable
     do_cmd "0" nostats enable
-    iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.2
-    ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.2
+    iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.2
+    ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.2
     diff $TESTTMP/save.1 $TESTTMP/save.2 || {
         echo "ip(6)tables-restore different for '$i'"
         exit 1
Index: ufw-0.36/tests/root/valid/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/valid/runtest.sh
+++ ufw-0.36/tests/root/valid/runtest.sh
@@ -30,22 +30,22 @@ echo "TESTING ARGS (default)" >> $TESTTM
 do_cmd "0"  enable
 do_cmd "0"  default allow
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  default deny
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  DEFAULT ALLOW
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  DEFAULT DENY
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 
Index: ufw-0.36/tests/root/valid6/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/valid6/runtest.sh
+++ ufw-0.36/tests/root/valid6/runtest.sh
@@ -35,41 +35,41 @@ do_cmd "0"  enable
 do_cmd "0"  default allow
 echo "ipv4:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 echo "ipv6:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    ip6tables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    ip6tables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  default deny
 echo "ipv4:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 echo "ipv6:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    ip6tables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    ip6tables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  DEFAULT ALLOW
 echo "ipv4:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 echo "ipv6:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    ip6tables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    ip6tables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 do_cmd "0"  DEFAULT DENY
 echo "ipv4:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    iptables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    iptables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 echo "ipv6:" >> $TESTTMP/result
 for i in INPUT FORWARD OUTPUT ; do
-    ip6tables -n -L $i | grep 'policy ' >> $TESTTMP/result
+    ip6tables -n -L $i 2>/dev/null | grep 'policy ' >> $TESTTMP/result
 done
 grep -h "DEFAULT" $TESTPATH/etc/default/ufw >> $TESTTMP/result
 
Index: ufw-0.36/tests/root/logging/runtest.sh
===================================================================
--- ufw-0.36.orig/tests/root/logging/runtest.sh
+++ ufw-0.36/tests/root/logging/runtest.sh
@@ -35,12 +35,12 @@ for i in allow deny limit reject ; do
         cat $TESTCONFIG/user.rules >> $TESTTMP/result
         cat $TESTCONFIG/user6.rules >> $TESTTMP/result
 
-        iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.1
-        ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.1
+        iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.1
+        ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.1
         do_cmd "0" nostats disable
         do_cmd "0" nostats enable
-        iptables-save | egrep -v '^(#|:)' > $TESTTMP/save.2
-        ip6tables-save | egrep -v '^(#|:)' >> $TESTTMP/save.2
+        iptables-save 2>/dev/null | egrep -v '^(#|:)' > $TESTTMP/save.2
+        ip6tables-save 2>/dev/null | egrep -v '^(#|:)' >> $TESTTMP/save.2
         diff $TESTTMP/save.1 $TESTTMP/save.2 || {
             echo "ip(6)tables-restore different for '$i'"
             exit 1
